server/src/test/java/org/elasticsearch/search/aggregations/metrics/MinAggregatorTests.java

```diff
@@ -43,183 +44,534 @@
+    public void testFormatter() throws IOException {
+        MinAggregationBuilder aggregationBuilder = new MinAggregationBuilder("min").field("number").format("0000.0");
+
+        MappedFieldType fieldType = new NumberFieldMapper.NumberFieldType(NumberFieldMapper.NumberType.INTEGER);
+        fieldType.setName("number");
 
-        IndexReader indexReader = DirectoryReader.open(directory);
-        IndexSearcher indexSearcher = newSearcher(indexReader, true, true);
+        testCase(aggregationBuilder, new MatchAllDocsQuery(), iw -> {
+            iw.addDocument(singleton(new NumericDocValuesField("number", 7)));
+            iw.addDocument(singleton(new NumericDocValuesField("number", 1)));
+        }, (Consumer<InternalMin>) min -> {
+            assertEquals(1.0, min.getValue(), 0);
+            assertTrue(AggregationInspectionHelper.hasValue(min));
+            assertEquals("0001.0", min.getValueAsString());
+        }, fieldType);
+    }
+
+    public void testGetProperty() throws IOException {
+        GlobalAggregationBuilder globalBuilder = new GlobalAggregationBuilder("global")
+            .subAggregation(new MinAggregationBuilder("min").field("number"));
 
-        MinAggregationBuilder aggregationBuilder = new MinAggregationBuilder("_name").field("number");
-        MappedFieldType fieldType = new NumberFieldMapper.NumberFieldType(NumberFieldMapper.NumberType.LONG);
+        MappedFieldType fieldType = new NumberFieldMapper.NumberFieldType(NumberFieldMapper.NumberType.INTEGER);
         fieldType.setName("number");
 
-        MinAggregator aggregator = createAggregator(aggregationBuilder, indexSearcher, fieldType);
-        aggregator.preCollection();
-        indexSearcher.search(new MatchAllDocsQuery(), aggregator);
-        aggregator.postCollection();
-        InternalMin result = (InternalMin) aggregator.buildAggregation(0L);
-        assertEquals(-1.0, result.getValue(), 0);
-        assertTrue(AggregationInspectionHelper.hasValue(result));
+        testCase(globalBuilder, new MatchAllDocsQuery(), iw -> {
+            iw.addDocument(singleton(new NumericDocValuesField("number", 7)));
+            iw.addDocument(singleton(new NumericDocValuesField("number", 1)));
+        }, (Consumer<InternalGlobal>) global -> {
+            assertEquals(1.0, global.getDocCount(), 2);
+            assertTrue(AggregationInspectionHelper.hasValue(global));
+            assertNotNull(global.getAggregations().asMap().get("min"));
+
+            InternalMin min = (InternalMin) global.getAggregations().asMap().get("min");
+            assertEquals(1.0, min.getValue(), 0);
+            assertThat(global.getProperty("min"), equalTo(min));
+            assertThat(global.getProperty("min.value"), equalTo(1.0));
+            assertThat(min.getProperty("value"), equalTo(1.0));
+        }, fieldType);
+    }
 
-        indexReader.close();
-        directory.close();
+    public void testSingleValuedFieldPartiallyUnmapped() throws IOException {
```

Comment:
Would be good to try this with a missing value set and make sure that gets picked up correctly too.